[libpq] build failure

**Host Environment**
- OS: Windows
- Compiler: Visual Studio 2019 16.10.2
- Port files: 6e024e744e7717c06ddacd5089401109c6298553

**To Reproduce**
Steps to reproduce the behavior:
`.\vcpkg\vcpkg.exe install libpq:x64-windows`

**Failure logs**
```
Computing installation plan...
The following packages will be built and installed:
    libpq[core,openssl,zlib]:x64-windows -> 12.2#16
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\m.karing\AppData\Local\vcpkg\archives\aa\aa6fb6fc8b1c8d3e2fdbfc43d2931de3974b9067.zip
Starting package 1/1: libpq:x64-windows
Building package libpq[core,openssl,zlib]:x64-windows...
-- Using cached C:/build/vcpkg/downloads/postgresql-12.2.tar.bz2
-- Cleaning sources at C:/build/vcpkg/buildtrees/libpq/src/tgresql-12-c160d9fcbb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/build/vcpkg/downloads/postgresql-12.2.tar.bz2
-- Applying patch patches/windows/install.patch
-- Applying patch patches/windows/win_bison_flex.patch
-- Applying patch patches/windows/openssl_exe_path.patch
-- Applying patch patches/windows/Solution.patch
-- Applying patch patches/windows/MSBuildProject_fix_gendef_perl.patch
-- Applying patch patches/windows/msgfmt.patch
-- Applying patch patches/windows/python_lib.patch
-- Applying patch patches/windows/fix-compile-flag-Zi.patch
-- Applying patch patches/windows/minimize_install.patch
-- Using source at C:/build/vcpkg/buildtrees/libpq/src/tgresql-12-c160d9fcbb.clean
-- Copying libpq source files to C:/build/vcpkg/buildtrees/libpq/x64-windows-rel...
-- Copying libpq source files... done
CMake Deprecation Warning at scripts/ports.cmake:16 (message):
  vcpkg_apply_patches has been deprecated in favor of the `PATCHES` argument
  to `vcpkg_from_*`.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_apply_patches.cmake:18 (z_vcpkg_deprecation_message)
  ports/libpq/portfile.cmake:127 (vcpkg_apply_patches)
  scripts/ports.cmake:142 (include)


-- Applying patch patches/windows/Solution_RELEASE.patch
-- Applying patch patches/windows/python3_build_RELEASE.patch
-- Patches applied!
-- Building libpq x64-windows-RELEASE...
-- Building libpq x64-windows-RELEASE... done
-- Building libecpg_compat x64-windows-RELEASE...
-- Building libecpg_compat x64-windows-RELEASE... done
-- Installing libpq x64-windows-RELEASE...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: C:/build/vcpkg/downloads/tools/perl/5.30.0.1/perl/bin/perl.exe install.pl C:/build/vcpkg/packages/libpq_x64-windows client
    Working Directory: C:/build/vcpkg/buildtrees/libpq/x64-windows-rel/src/tools/msvc
    Error code: 2
    See logs for more information:
      C:\build\vcpkg\buildtrees\libpq\install-x64-windows-RELEASE-err.log

Call Stack (most recent call first):
  ports/libpq/portfile.cmake:233 (vcpkg_execute_required_process)
  scripts/ports.cmake:142 (include)


Error: Building package libpq:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: libpq:x64-windows
  Vcpkg version: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9
```
[install-x64-windows-RELEASE-err.log](https://github.com/microsoft/vcpkg/files/6675755/install-x64-windows-RELEASE-err.log)

--- a/ports/libpq/patches/windows/install.patch
+++ b/ports/libpq/patches/windows/install.patch
@@ -2,6 +2,25 @@ diff --git a/src/tools/msvc/Install.pm b/src/tools/msvc/Install.pm
 index d109b2c90..75c7446e6 100644
 --- a/src/tools/msvc/Install.pm
 +++ b/src/tools/msvc/Install.pm
+@@ -63,8 +63,16 @@ sub Install
+ 		do "./config.pl" if (-f "config.pl");
+ 	}
+ 
+-	chdir("../../..")    if (-f "../../../configure");
+-	chdir("../../../..") if (-f "../../../../configure");
++	# Move to the root path depending on the current location.
++	if (-f "../../../configure")
++	{
++		chdir("../../..");
++	}
++	elsif (-f "../../../../configure")
++	{
++		chdir("../../../..");
++	}
++
+ 	my $conf = "";
+ 	if (-d "debug")
+ 	{
 @@ -106,7 +106,7 @@ sub Install
  			}
  		},

--- a/ports/libpq/vcpkg.json
+++ b/ports/libpq/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libpq",
   "version": "12.2",
-  "port-version": 17,
+  "port-version": 18,
   "description": "The official database access API of postgresql",
   "homepage": "https://www.postgresql.org/",
   "supports": "!uwp",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3478,7 +3478,7 @@
     },
     "libpq": {
       "baseline": "12.2",
-      "port-version": 17
+      "port-version": 18
     },
     "libpqxx": {
       "baseline": "7.3.1",

--- a/versions/l-/libpq.json
+++ b/versions/l-/libpq.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "03c934a8004ec6c34842748d18c06a38e7a8aa3e",
+      "version": "12.2",
+      "port-version": 18
+    },
+    {
       "git-tree": "01e2ad37b9df46e45826e52a6cddfb95ca137abd",
       "version": "12.2",
       "port-version": 17

